### PR TITLE
[#640] feat(netty): Metric system for netty server

### PR DESCRIPTION
**Pull request by xumanbu**
### What changes were proposed in this pull request?

add NettyMetrics for StreamSever 

### Why are the changes needed?

Fix: #640

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?
UT

### CODE REVIEW

**Comment by jerqi** [comment]
@leixm Could you help me review this pr?

**Comment by leixm** [comment]
> @leixm Could you help me review this pr?

Sure.

File: common/src/main/java/org/apache/uniffle/common/metrics/NettyMetrics.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  }
+
+  @Override
+  public void registerGeneralMetrics() {
+    gaugeNettyActiveConn = metricsManager.addLabeledGauge(NETTY_ACTIVE_CONNECTION);
+    counterNettyException = metricsManager.addLabeledCounter(NETTY_HANDLE_EXCEPTION);
+  }
+
+  public void incRequest(String className) {
+    if (isRegistered) {
```
Comment (leixm) [inline common/src/main/java/org/apache/uniffle/common/metrics/NettyMetrics.java:42]: Could we reuse RPCMetrics#incCounter? seems same code logic.

File: common/src/main/java/org/apache/uniffle/common/metrics/NettyMetrics.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      }
+      Counter.Child counter = counterMap.get(className);
+      if (counter != null) {
+        counter.inc();
+      }
+    }
+  }
+
+  public void decRequest(String className) {
+    if (isRegistered) {
```
Comment (leixm) [inline common/src/main/java/org/apache/uniffle/common/metrics/NettyMetrics.java:55]: ditto.

**Review by leixm — CHANGES_REQUESTED** [review]

File: common/src/main/java/org/apache/uniffle/common/metrics/NettyMetrics.java (hunk 1 above)
Comment (xumanbu) [reply common/src/main/java/org/apache/uniffle/common/metrics/NettyMetrics.java:42]: good advice.

**Comment by leixm** [comment]
+1 LGTM.

**Review by leixm — APPROVED** [review]

**Review by jerqi — APPROVED** [review]
LGTM.

**Pull request merged**